### ISSUE TITLE
DC cannot access files in this path

### ISSUE THREAD

**Issue by ags1234** [doublecmd/doublecmd#226]
I have the following path:

`d:\work\2021\2021 Quality control analysis 1\2021.10.25 Things to check for in the reports of employees\2.quick brown fox jumps over the lazy dog the 02 (brown fox jumps)\Specimen_AAAA\2021-10-25_102558240_abcd11_efgh04_8715000_0b7ceab4-5622-4159-9faf-a2e29dd1a4e6.jpg`

I use doublecmd-1.1.0.r9938.i386-win32, in Windows XP SP3, 32 bit.

I know... it is long. I can use DC to traverse all the folders, but cannot access the jpg files, nor perform any operations on them. All of them are inaccessible, with a question mark icon. Using Total Commander or Windows Explorer I could access the files.

[Edit: some operations can be performed, see posts below]

**Comment by ags1234** [comment]
I just encountered this problem again, trying to launch a video file from a 200 character long path. The video player launches, but then it says the file cannot be found.

In TC, with the same path, same file and video player, the player launches, and the file plays correctly.

**Comment by RamonUnch** [comment]
You mean you double-click directly in DC?
Do you get the same problem drag&dropping from DC in the video player?
From what yo say 200char long path + 1 backslash + 60chars = 261 chars (=>too long to handle doubleclick/Contect menu on DC)
For me as long as the path 259 chars or shorter everything works fine.
TC (from what I remember) try using the short path when the total file+path length exceeds the 259 limit, this is helpful for non long path aware programs.
Some media players such as MPlayer do handle long path but you need to pass him the '\\?\' prefixed path, which is not so simple to do.

Maybe I understand wrong could you give a precise path+filemanme example that does this?

**Comment by ags1234** [comment]
@RamonUnch: yes, double-click directly in DC. Your comment and the fact that I had some time, determined me analyze the problem a little more.

In my long path, I have a few video files. Digging deeper into the problem, I just noticed that only one of them did not play. The full path is like below (I had to edit out only the names, the rest is identical. It is a course I have to take.

```c:\zzz\aaaaaa-2.87-bbbbbbbb\ccccccccc\[ dddddddd.bbb] eeeee - fffffffff gggggggggg - aaa bbb cc ddddddddddd !\_aaa bbbb ccccc dddd !\1. aaa bbbbbbbbbb ccccccccccc dd ee fffffffff  [fffffffff eeeeeee]\2. The Starting Point - This is where everything begins!.mp4```

Strangely enough, today the file played fine. But in the player the file name appears like '2THEST~1.MP4'.

**Comment by RamonUnch** [comment]
@ags1234 OK, I think I understand what DC actually does. Your path is exactly 259 char long = MAX_PATH-1.
DC seems to always add the '\\?\' prefix to the file name when calling some specific programs eg: `Notepad`. On newer windows Notepad does handle long path names, so it seems to be a good idea. However if notepad did not handle long path names this would not be a good idea.

So let us imagine that DC for some reasons assume that your media player (which is?) does handle long path names. In this case upon double clicking DC will call: `mediaplayer.exe "\\?\c:\zzz\aaaaaa-c.dd-bbbbbbbb\ccccccccc\[ dddddddd.bbb] eeeee - fffffffff gggggggggg - aaa bbb cc ddddddddddd !\_aaa bbbb ccccc dddd !\1. aaa bbbbbbbbbb ccccccccccc dd ee fffffffff [fffffffff eeeeeee]\2. The Starting Point - This is where everything begins!.mp4"`
Now let us imagine that the `mediaplayer.exe` does not actually handle long path names. In this case the path length to consider that must be smaller or equal to 259 includes the `\\?\` prefix. So the new file length is 4 chars longer than the original path and the new limit becomes 255 for the non prefixed path.
Probably Your media player uses the short name if the path is too long and that is why you see the short file name in the title bar.

So DC should probably avoid using UNC path unless the full path name exceeds 259.

**Comment by RamonUnch** [comment]
Actually I was wrong, the '\\?\' prefix was here because of some old reg tweak I did some time ago (not related to DC). So I am not sure what happens. You can always try to shorten the path to 255 and see if it works properly. DC did nothing wrong from what I tried.

**Comment by ags1234** [comment]
My audio player in mplayer2.exe from Windows XP SP3. For the moment, I am restricted to upgrade to a newer operating system.

The \\?\ prefix is a method of addressing devices. Look [here](https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats#dos-device-paths).
